lib/redis_memo/memoizable/dependency.rb

```diff
@@ -40,6 +43,20 @@ def depends_on(dependency, **conditions)
     end
   end
 
+  def extract_dependencies_for_relation(relation)
+    connection = ActiveRecord::Base.connection
+    query, binds = connection.send(:to_sql_and_binds, relation.arel) # Is there another way to get this info?
+    RedisMemo::MemoizeQuery::CachedSelect.current_query = relation.arel
+    is_query_cached = RedisMemo::MemoizeQuery::CachedSelect.extract_bind_params(query)
+      raise(
+        RedisMemo::ArgumentError,
+        "Invalid Arel dependency. Query is not enabled for RedisMemo caching."
+      ) unless is_query_cached
+      extracted_dependency = connection.dependency_of(:exec_query, query, nil, binds)
+  ensure
+    RedisMemo::MemoizeQuery::CachedSelect.reset_current_query
```

Comment:
> Is it safe to also hijack the thread local variables that are shared between activerecord/redismemo in this case?

I think there might be an issue: this assumes people won't run actual queries (exec_query) in the `depends_on` which might not be true. If there is an `exec_query` while extracting dependencies, it might continue to use whatever thread locals we have set since we have not cleared them yet. We can probably just reset everything before calling dependency_of tho